client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -625,19 +631,28 @@ function getNoticeType( cart: ResponseCart | undefined ): Record< string, string
 }
 
 function getUrlWithQueryParam( url: string, queryParams: Record< string, string > ): string {
-	const { protocol, hostname, port, pathname, query, hash } = parseUrl( url, true );
-
-	return formatUrl( {
-		protocol,
-		hostname,
-		port,
-		pathname,
-		query: {
-			...query,
+	const defaultUrl = '/';
+	url = url || defaultUrl;
```

Comment:
It's always better to avoid mutating the input variable/argument to avoid unexpected results/bugs.

Reply:
Yes you're right, thank you. 👍 - Fixed.